build-challenge/tsconfig.json

```diff
@@ -1,8 +1,8 @@
 { 
     "compilerOptions":{ 
        "target":"es6",
-       "moduleResolution":"node",
-       "module":"es2020",
+       "moduleResolution":"NodeNext",
+       "module": "NodeNext",
```

Comment:
I have a preference for this to be es2020 since it more accurately aligns with what the JS engine in minecraft supports, at least today

Reply:
Same for other tsconfig.json changes

Reply:
Fixed across all the files